examples/ogbn_products_revgnn.py

```diff
@@ -0,0 +1,271 @@
+            self.dropout.set_mask(dropout_mask)
+            out = self.dropout(out)
+
+        out = self.gnn(out, edge_index)
+
+        return out
+
+
+class GNNBlock(BasicBlock):
+    def __init__(self, in_channels, out_channels, aggr="mean"):
+        super().__init__(in_channels)
+        self.gnn = SAGEConv(in_channels, out_channels, aggr=aggr)
+
+
+class RevGNN(torch.nn.Module):
+    """Group Reversible GNN proposed in GNN1000:
+    https://arxiv.org/abs/2106.07476."""
+    def __init__(
+        self,
+        in_channels,
+        hidden_channels,
+        out_channels,
+        num_layers,
+        dropout,
+        num_groups=2,
+        aggr="mean",
+    ):
+        super().__init__()
+
+        self.num_layers = num_layers
+        self.dropout = dropout
+
+        self.gnns = torch.nn.ModuleList()
+        self.last_norm = LayerNorm(hidden_channels, elementwise_affine=True)
+
+        self.node_features_encoder = Linear(in_channels, hidden_channels,
+                                            bias=True,
+                                            weight_initializer="glorot")
+        self.node_pred_linear = Linear(
+            hidden_channels,
+            out_channels,
+            bias=True,
+            weight_initializer="glorot",
+        )
+
+        assert (hidden_channels // num_groups) * num_groups == hidden_channels
+        for _ in range(self.num_layers):
+            seed_gnn = GNNBlock(
+                hidden_channels // num_groups,
+                hidden_channels // num_groups,
+                aggr,
+            )
+            gnn = GroupAddRev(seed_gnn, num_groups=num_groups)
+
+            self.gnns.append(gnn)
+
+    def forward(self, x, edge_index):
+        h = self.node_features_encoder(x)
+
+        m = torch.zeros_like(h).bernoulli_(1 - self.dropout)
+        mask = m.requires_grad_(False) / (1 - self.dropout)
+
+        for layer in range(self.num_layers):
+            h = self.gnns[layer](h, edge_index, mask)
+
+        h = F.relu(self.last_norm(h))
+        h = F.dropout(h, p=self.dropout, training=self.training)
+
+        h = self.node_pred_linear(h)
+
+        return torch.log_softmax(h, dim=-1)
+
+
+root = osp.join(osp.dirname(osp.realpath(__file__)), "..", "data", "products")
+dataset = PygNodePropPredDataset("ogbn-products", root)
+split_idx = dataset.get_idx_split()
+evaluator = Evaluator(name="ogbn-products")
+data = dataset[0]
+data.edge_index = add_self_loops(edge_index=data.edge_index,
+                                 num_nodes=data.num_nodes)[0]
+data.y.squeeze_()
+
+# Set split indices to masks.
+for split in ["train", "valid", "test"]:
+    split_mask = torch.zeros(data.num_nodes, dtype=torch.bool)
+    split_mask[split_idx[split]] = True
+    data[f"{split}_mask"] = split_mask
+
+train_loader = RandomNodeSampler(data, num_parts=10, shuffle=True,
+                                 num_workers=5)
+test_loader = RandomNodeSampler(data, num_parts=1, num_workers=5)
```

Comment:
Why do we need the `RandomNodeSampler` here (since `num_parts=1`)?

Reply:
Inference on the full batch graph may take about 20G of GPU memory. Just in case someone has a smaller GPU and this could be handy for one to increase the `num_parts`. Added some comments.